docker/templates/base/latest/Dockerfile

```diff
@@ -1,107 +1,116 @@
-# Base image with Tomcat and KPM
-FROM ubuntu:16.04
-MAINTAINER Kill Bill core team <killbilling-users@googlegroups.com>
+FROM ubuntu:20.04 as builder-base
+LABEL maintainer="killbilling-users@googlegroups.com"
 
 USER root
 
-ENV LC_CTYPE en_US.UTF-8
-ENV LC_ALL C
-ENV LANG en_US.UTF-8
-ENV PYTHONIOENCODING utf8
+ENV LC_CTYPE=en_US.UTF-8
+ENV LC_ALL=C
+ENV LANG=en_US.UTF-8
+ENV PYTHONIOENCODING=utf8
 
-# Install Kill Bill dependencies and useful tools
-ENV DEBIAN_FRONTEND noninteractive
+# Install convenient utilities
+# https://github.com/moby/moby/issues/4032
+ARG DEBIAN_FRONTEND=noninteractive
 RUN apt-get update && \
-    apt-get install -y \
+    apt-get install --no-install-recommends -y \
+      ansible \
+      # https://github.com/tianon/docker-brew-ubuntu-core/issues/59
       apt-utils \
       curl \
-      dpkg-dev \
-      gcc \
       git \
       less \
       libapr1 \
-      libapr1-dev \
-      libssl-dev \
-      make \
       mysql-client \
       net-tools \
-      openjdk-8-jdk \
-      python-pip \
+      openjdk-8-jdk-headless \
+      python3-lxml \
       sudo \
       telnet \
       unzip \
+      # A bit large unfortunately (https://bugs.launchpad.net/ubuntu/+source/vim/+bug/1884583)
```

Comment:
But will `vim` (and other utilities really be part of the image in the end, considering we use the 2 step approach `FROM builder-base as builder-tomcat`, and in the second step only install a minimal amount of tools?

Reply:
There is a third step :-)

```
# Final base image with Tomcat and KPM
FROM builder-base
```